test/testCFile.js

```diff
@@ -0,0 +1,660 @@
+            type: cft
+        });
+        test.ok(cf);
+
+        cf.parse("const char* pPicture = resBundle_getLocStringWithKey(resBundle,'PictureMode.Standard', 'Standard');");
+
+        var set = cf.getTranslationSet();
+        test.ok(set);
+
+        var r = set.getBy({
+            reskey: "PictureMode.Standard"
+        });
+        test.ok(r);
+        test.equal(r[0].getSource(), "PictureMode.Standard");
+        test.equal(r[0].getKey(), "PictureMode.Standard");
+
+        test.done();
+    },
+
+    testCFileParseMultiple: function(test) {
+        test.expect(8);
+
+        var cf = new CFile({
+            project: p,
+            pathName: undefined,
+            type: cft
+        });
+        test.ok(cf);
+
+        cf.parse('char *screen_share_60 = (char *)resBundle_getLocString(res_bundle, "Block");\n\ta.parse("This is another test.");\n\t\tchar *screen_share_67 = (char *)resBundle_getLocString(res_bundle, "Cancel");');
+
+        var set = cf.getTranslationSet();
+        test.ok(set);
+
+        var r = set.getBySource("Block");
+        test.ok(r);
+        test.equal(r.getSource(), "Block");
+        test.equal(r.getKey(), "Block");
+
+        r = set.getBySource("Cancel");
+        test.ok(r);
+        test.equal(r.getSource(), "Cancel");
+        test.equal(r.getKey(), "Cancel");
+
+        test.done();
+    },
+
+    testCFileParseMultipleWithKey: function(test) {
+        test.expect(10);
+
+        var cf = new CFile({
+            project: p,
+            pathName: undefined,
+            type: cft
+        });
+        test.ok(cf);
+
+        cf.parse('char *screen_share_60 = (char *)resBundle_getLocStringWithKey(res_bundle, "Block.key", "Block");\n\ta.parse("This is another test.");\n\t\tchar *screen_share_67 = (char *)resBundle_getLocStringWithKey(res_bundle, "Cancel.key","Cancel");');
+
+        var set = cf.getTranslationSet();
+        test.ok(set);
+
+        var r = set.getBy({
+            reskey: "Block.key"
+        });
+        test.ok(r);
+        test.equal(r[0].getSource(), "Block.key");
+        test.ok(r[0].getAutoKey());
+        test.equal(r[0].getKey(), "Block.key");
+
+        r = set.getBy({
+            reskey: "Cancel.key"
+        });
+        test.ok(r);
+        test.equal(r[0].getSource(), "Cancel.key");
+        test.ok(r[0].getAutoKey());
+        test.equal(r[0].getKey(), "Cancel.key");
+
+        test.done();
+    },
+
+    testCFileParseMultipleSameLine: function(test) {
+        test.expect(12);
+
+        var cf = new CFile({
+            project: p,
+            pathName: undefined,
+            type: cft
+        });
+        test.ok(cf);
+
+        cf.parse('char *screen_share_60 = (char *)resBundle_getLocStringWithKey(res_bundle, "Block.key", "Block");\n\ta.parse("This is another test.");\n\t\tchar *screen_share_67 = (char *)resBundle_getLocString(res_bundle, "Cancel"); \n\tchar *screen_share_70 = (char *)resBundle_getLocString(res_bundle, "Stop");');
```

Comment:
These are on different lines (with a \n in the middle). The point of testing on the same line is to make sure that the regular expression can work properly with multiple strings to extract on the same line.

Reply:
This one will still fail if you remove the \n in the parsed string. The regular expressions are not correct yet.

Reply:
I have additional test cases and it seems fine. Did I get it wrong?; ;(
https://github.com/iLib-js/ilib-loctool-webos-c/blob/initialSetup/test/testCFile.js#L620
https://github.com/iLib-js/ilib-loctool-webos-c/blob/initialSetup/test/testCFile.js#L654https://github.com/iLib-js/ilib-loctool-webos-c/blob/initialSetup/test/testCFile.js#L689